pkg/sql/pg_catalog.go

```diff
@@ -1107,9 +1108,71 @@ var pgCatalogDefaultACLTable = virtualSchemaTable{
 https://www.postgresql.org/docs/9.6/catalog-pg-default-acl.html`,
 	schema: vtable.PGCatalogDefaultACL,
 	populate: func(ctx context.Context, p *planner, dbContext catalog.DatabaseDescriptor, addRow func(...tree.Datum) error) error {
+		h := makeOidHasher()
+		if dbContext.GetDefaultPrivileges() != nil {
+			for _, defaultPrivs := range dbContext.GetDefaultPrivileges().DefaultPrivileges {
+				// Need to consider the case of USAGE for Public as well.
+				for objectType, privs := range defaultPrivs.DefaultPrivilegesPerObject {
+					// Type of object this entry is for:
+					// r = relation (table, view), S = sequence, f = function, T = type, n = schema.
+					var c string
+					switch objectType {
+					case tree.Tables:
+						c = "r"
+					case tree.Sequences:
+						c = "S"
+					case tree.Types:
+						c = "T"
+					case tree.Schemas:
+						c = "n"
+					}
+					privilegeObjectType := targetObjectToPrivilegeObject[objectType]
+					arr := tree.NewDArray(types.String)
+					for _, userPrivs := range privs.Users {
+						var user string
+						if userPrivs.UserProto.Decode().IsPublicRole() {
+							// Postgres represents Public in defacl as an empty string.
+							user = ""
+						} else {
+							user = userPrivs.UserProto.Decode().Normalized()
+						}
+
+						defaclItem := fmt.Sprintf(`%s=%s/%s`,
+							user,
+							privilege.PrivilegeListToACL(
+								privilege.ListFromBitField(userPrivs.Privileges, privilegeObjectType),
+								privilegeObjectType,
+							),
+							"", /* grantor, currently we don't track grantor. */
```

Comment:
would you mind adding a comment pointing at https://github.com/cockroachdb/cockroach/issues/67442